### ISSUE TITLE
Write an example psbt workflow

### ISSUE THREAD

**Issue by sanket1729** [rust-bitcoin/rust-bitcoin#892]
Write an example in the `examples` directory with the workflow from [BIP174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki#test-vectors). See the valid example that starts from the line in the above link.
```
The private keys in the tests below are derived from the following master private key:
```
- Use BIP32 wallet APIs to derive all priv/public keys
- Use the sighash APIs to compute the signature hashes
- Use all Psbt supported APIs
- Identify the API pain-points and suggest improvements.

**Comment by caglaryucekaya** [comment]
Hi I'd like to work on this

**Comment by sanket1729** [comment]
Feel free to ask any questions if you struggle somewhere

**Comment by DanGould** [comment]
I started working on this here https://github.com/DanGould/rust-bitcoin/tree/example/psbt, I'd like to draft a PR.

@caglaryucekaya are you still working on this?

**Comment by caglaryucekaya** [comment]
No I got unexpectedly busy after writing this, you can take it

**Comment by tcharding** [comment]
Hey @DanGould, cool to see you working on this, I had a look at your tree. I've been messing around with this too, I'm happy to review your code if you put up a PR. I learned some things from you version already though so perhaps we can push this over the line together, its an interesting issue I've learned a bunch about PSBTs already :) I'm stuck on finalizing the PSBT, I've got the legacy input done but can't work out how to build the witness stack for the other input. Anyways, if you are interested my code is on a branch called `psbt-example`. Cheers.
